### PR TITLE
Split providers and grant config to a new file

### PR DESCRIPTION
**Pull request by abdonrd**
### What does it do?

Split providers and grant config to a new file

### Why is it needed?

To create new providers more easily.

### CODE REVIEW

**Review by alexandrebodin — CHANGES_REQUESTED** [review]
Hi @abdonrd thanks for this cleanup. Unfortunately this will break our documentation on how to contribute to this. Can you make an update and check with @derrickmehaffy ?

**Comment by abdonrd** [comment]
@alexandrebodin @derrickmehaffy new PR for the documentation! https://github.com/strapi/strapi/pull/8794

File: packages/strapi-plugin-users-permissions/services/Providers.js [hunk 1]
```diff
@@ -6,13 +6,9 @@
...
 // Public node modules.
 const _ = require('lodash');
-const request = require('request');
 
-// Purest strategies.
-const purest = require('purest')({ request });
-const purestConfig = require('@purest/providers');
 const { getAbsoluteServerUrl } = require('strapi-utils');
-const jwt = require('jsonwebtoken');
+const providers = require('./ProviderList');
```
Comment (alexandrebodin) [inline packages/strapi-plugin-users-permissions/services/Providers.js:11]: can you please use kebab-case-naming for the new file :)

File: packages/strapi-plugin-users-permissions/services/Providers.js [hunk 2]
```diff
@@ -123,351 +119,10 @@ const getProfile = async (provider, query, callback) => {
...
-      } catch (err) {
-        callback(err);
-      }
-      break;
-    }
-    default:
-      callback(new Error('Unknown provider.'));
-      break;
+  if (providers[provider]) {
+    await providers[provider](grant, access_token, query, callback);
```
Comment (alexandrebodin) [inline packages/strapi-plugin-users-permissions/services/Providers.js:123]: can you pass an object instead 

```js
const context = { grant, access_token, query }
providers[provider](context, callback)
```

**Review by alexandrebodin — CHANGES_REQUESTED** [review]

**Comment by abdonrd** [comment]
@alexandrebodin thanks for the review!

I created another PR (https://github.com/strapi/strapi/pull/8942) because this appears broken for a reason that I do not understand:

Screenshot 2020-12-18 at 17 48 35

**Pull request closed without merging**